src/layers/sequential.rs

```diff
@@ -60,3 +61,32 @@ where
         Tensor::from_shape_vec(output_shape, outputs).unwrap()
     }
 }
+
+impl<D: Dimension> Model for Sequential<D> {
+    fn state_dict(&self) -> IndexMap<String, Vec<f64>> {
+        let mut state_dict: IndexMap<String, Vec<f64>> = IndexMap::new();
+        let layers_iter = self.layers.iter();
+
+        for layer in layers_iter.filter(|x| x.is_trainable()) {
+            let weights_key = format!("{}{}", layer.name(), ".weight");
+            let biases_key = format!("{}{}", layer.name(), ".bias");
+
+            state_dict.insert(weights_key, layer.weights().clone());
+            state_dict.insert(biases_key, layer.biases().clone());
+        }
```

Comment:
Might as well do the filtering in the same line as you converted the vector into an iterator. Also you don't need to clone weights and biases before inserting.
```suggestion
        let trainable_layers = self.layers.iter().filter(|x| x.is_trainable());

        for layer in trainable_layers {
            let weights_key = format!("{}{}", layer.name(), ".weight");
            let biases_key = format!("{}{}", layer.name(), ".bias");

            state_dict.insert(weights_key, layer.weights());
            state_dict.insert(biases_key, layer.biases());
        }
```